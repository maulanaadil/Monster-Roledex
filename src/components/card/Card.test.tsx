import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';
import 'jest-styled-components';

describe('renders a Card', () => {
  const initProps = {
    id: 1,
    name: 'test',
    email: 'test@test.com',
  };

  const wrapper = shallow(<Card {...initProps} />);
  afterAll(() => wrapper.unmount());

  it('should render a card components if props presents', () => {
    expect(wrapper).toHaveStyleRule('background-color', '#95dada');
    expect(wrapper.find('h2').text()).toContain(initProps.name);
    expect(wrapper.find('p').text()).toContain(initProps.email);
    expect(wrapper.find('img').prop('src')).toBe(
      `https://robohash.org/${initProps.id}?set=set2&size=180x180`
    );
  });

  it('should render same snapshots of card component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
