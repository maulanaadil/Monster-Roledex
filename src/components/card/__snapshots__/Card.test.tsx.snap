// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Card should render same snapshots of card component 1`] = `
<styled.div>
  <img
    alt="monster"
    src="https://robohash.org/1?set=set2&size=180x180"
  />
  <h2>
    test
  </h2>
  <p>
    test@test.com
  </p>
</styled.div>
`;
